Cabal/src/Distribution/Simple/Haddock.hs

```diff
@@ -714,10 +816,127 @@ fromExecutable verbosity haddockArtifactsDirs lbi clbi htmlTemplate exe = do
       htmlTemplate
       inFiles
       (buildInfo exe)
+  let args' =
+        commonArgs
+          <> args
+            { argOutputDir =
+                Dir $
+                  haddockDirName haddockTarget pkg_descr
+                    </> unUnqualComponentName (exeName exe)
+            }
   return
-    args
-      { argOutputDir = Dir $ unUnqualComponentName $ exeName exe
-      , argTitle = Flag $ unUnqualComponentName $ exeName exe
+    args'
+      { argTitle = Flag $ unUnqualComponentName $ exeName exe
+      , -- we need to accommodate for `argOutputDir`
```

Comment:
```suggestion
      , -- we need to accommodate `argOutputDir`
```
It's either "account for" or "accommodate".

Reply:
Thanks @geekosaur.